src/balena.sbvr

```diff
@@ -783,6 +807,7 @@ Rule: It is necessary that each release that has a release version1 and has a st
 Rule: It is necessary that each image that has a status that is equal to "success", has a push timestamp.
 Rule: It is necessary that each application that is public, has an application type that is not legacy.
 Rule: It is necessary that each application that owns a release1 that has a status that is equal to "success" and has a commit1, owns at most one release2 that has a status that is equal to "success" and has a commit2 that is equal to the commit1.
+Rule: It is necessary that each application that owns a release1 that has a revision, owns at most one release2 that has a semver major that is of the release1 and has a semver minor that is of the release1 and has a semver patch that is of the release1 and has a revision that is of the release1.
```

Comment:
Would you prefer it this way v?
```sbvr
It is necessary that each application that owns a release1 that has a **revision1**, owns at most one release2 that has a semver major that is of the release1 and has a semver minor that is of the release1 and has a semver patch that is of the release1 and has a revision2 **that is is equal to the revision1**.
```
The `release1 that has a revision` is there in order to filter out all releases that are draft upfront, but then I decided to not name it and use `that is of release1` for consistency with the rest semver major/minor/patch checks (which also just use `that is of release1` in order to avoid having all parts of release1 and make the rule a bit shorter).

The full rule which would consistently use `equals` instead of `is of` would be huge:
```sbvr
It is necessary that each application that owns a release1 that has a revision1 and has a semver major1 and has a semver minor1 and has a semver patch1, owns at most one release2 that has a semver major2 that is equal to the semver major1 and has a semver minor2 that is equal to the semver minor1 and has a semver patch2 that equal to the semver patch1 and has a revision2 that is is equal to the revision1.
```

Reply:
Thanks for the explanation Thodori. Your new version reads a bit better. Honestly, if I haven't seen the SQL / TS code and your comments, I probably wouldn't understand this rule.

> The ideal would be if pine supported using computed terms in rules

Right. Indeed, such feature would help here.

Reply:
TBH I prefer the consistency rather than mixing `is of` and `is equal to` :thinking: Let's see whether Page has any preference on this.

Reply:
With the index already in place, this takes just 30ms :)